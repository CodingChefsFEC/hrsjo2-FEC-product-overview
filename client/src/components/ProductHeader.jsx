import React from 'react';
import {ProductHeaderStyles} from '../styles/App.styles.js';

const ProductHeader = ({productData}) => {
  const { name, brand, breadcrumbs } = productData;
  return (
    <ProductHeaderStyles>
      <div className="productHeader">
        <div className="productHeader--breadcrumbs">
          {breadcrumbs && breadcrumbs.map(crumb => {
            return (
              //TODO add unique key
              <span key={crumb}>
                <a href="#" className="productHeader--crumb">{crumb}</a>
                <span>/</span>
              </span>
            )
          })}
        </div>
        <h1 className="productHeader--productName">{name}</h1>
        <a href="#" className="productHeader--brandLink">{`Shop all ${brand}`}</a>
      </div>
    </ProductHeaderStyles>

  )
}

export default ProductHeader;