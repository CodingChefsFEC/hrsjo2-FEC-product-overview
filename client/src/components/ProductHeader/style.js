import styled from "styled-components";

const targetRed = 'rgb(204, 0, 0)';
const lightGray = 'rgb(102, 102, 102)';
const darkGray = 'rgb(51, 51, 51)';
const darkGreen = 'rgb(0, 102, 1)';
const warningOrange = 'rgb(184, 83, 0)';

export const Breadcrumbs = styled.div`
    color: ${lightGray};
    font-size: 12px;
    padding: 4px 4px;

    .divider {
        margin: 0 0.2rem;
    }
`;

export const ProductName = styled.h1`
    font-size: 23px;
    line-height: 1.25;
    margin-top: 4px;
    margin-bottom: 4px;
`;