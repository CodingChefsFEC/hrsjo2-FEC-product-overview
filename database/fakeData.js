const fakeData = JSON.parse(`{
  "_id": {
    "$oid": "5fae237faafe544c4495f638"
  },
  "breadcrumbs": [
    "Target",
    "Bikes",
    "Chipss"
  ],
  "images": [
    "http://placeimg.com/640/480/food",
    "http://placeimg.com/640/480/abstract",
    "http://placeimg.com/640/480/fashion"
  ],
  "id": 0,
  "name": "Gorgeous Frozen Car",
  "brand": "Bergnaum Group",
  "price_reg": 50.37,
  "price_discount": 415.43,
  "sale_end": {
    "$date": "2021-08-27T12:13:01.224Z"
  },
  "total_reviews": 112,
  "total_questions": 6,
  "ratings_data": {
    "_id": {
      "$oid": "5fae237faafe544c4495f639"
    },
    "five_star": 96,
    "four_star": 9,
    "three_star": 12,
    "two_star": 33,
    "one_star": 39
  },
  "promo_data": {
    "_id": {
      "$oid": "5fae237faafe544c4495f63a"
    },
    "id": 4,
    "icon_img": "http://placeimg.com/640/480",
    "main_text": "Sit quo aut distinctio veniam.",
    "sub_text": "Sed nulla nihil modi non quia eligendi quia dignissimos.",
    "link_text": "Learn More"
  },
  "__v": 0
}`);

export default fakeData;