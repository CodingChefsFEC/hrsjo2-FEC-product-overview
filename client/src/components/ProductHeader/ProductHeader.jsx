import React from 'react';
import { Breadcrumbs, ProductName } from './style.js'


const ProductHeader = ({productData}) => {
  const { name, brand, breadcrumbs } = productData;
  return (
      <div className="productHeader">
        <Breadcrumbs>
          {breadcrumbs && breadcrumbs.map(crumb => {
            return (
              //TODO add unique key
              <span key={crumb}>
                <a href="#" className="productHeader--crumb">{crumb}</a>
                <span>/</span>
              </span>
            )
          })}
        </Breadcrumbs>
        <ProductName>{name}</ProductName>
        <a href="#" className="productHeader--brandLink">{`Shop all ${brand}`}</a>
      </div>

  )
}

export default ProductHeader;