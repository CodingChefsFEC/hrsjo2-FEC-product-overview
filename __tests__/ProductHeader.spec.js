import React from 'react';
import { create } from 'react-test-renderer';
import ProductHeader from '../client/src/components/ProductHeader/ProductHeader.jsx';
import fakeData from '../database/fakeData.js';

// Snapshot test
describe("ProductHeader component", () => {
  test("Matches the snapshot", () => {
    const productHeader = create(<ProductHeader productData={fakeData} />);
    expect(productHeader.toJSON()).toMatchSnapshot();
  })
});

