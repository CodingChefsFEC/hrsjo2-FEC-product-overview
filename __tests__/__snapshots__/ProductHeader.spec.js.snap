// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductHeader component Matches the snapshot 1`] = `
<div
  className="productHeader"
>
  <div
    className="style__Breadcrumbs-sc-8km5h0-0 bLEPZT"
  >
    <span>
      <a
        className="productHeader--crumb"
        href="#"
      >
        Target
      </a>
      <span>
        /
      </span>
    </span>
    <span>
      <a
        className="productHeader--crumb"
        href="#"
      >
        Bikes
      </a>
      <span>
        /
      </span>
    </span>
    <span>
      <a
        className="productHeader--crumb"
        href="#"
      >
        Chipss
      </a>
      <span>
        /
      </span>
    </span>
  </div>
  <h1
    className="style__ProductName-sc-8km5h0-1 fOviLX"
  >
    Gorgeous Frozen Car
  </h1>
  <a
    className="productHeader--brandLink"
    href="#"
  >
    Shop all Bergnaum Group
  </a>
</div>
`;
